Flow ➸ TypeScript

rye opened the pull request:
As we look towards TypeScript as a possible replacement (to make React-Native upgrades a bit better), this is one key step.

Here's the general process I followed:

- Rename `.js` files to `.ts`, or if they contain JSX things, rename them to `.tsx`.
- Run `yarn run p`
- If any parsing errors, fix them
- Run `yarn run lint` and fix or disable any lints (didn't really have to disable that many lints)
- Run `yarn run bundle:android` to ensure that we can actually generate a legit bundle (there are some parsing errors that prettier doesn't hit, but the bundler does, for some reason)

rye committed:
ci: Disable flow check

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
package: Remove "flow" script

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
danger: Remove mentions of flow and flow-related tasks

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
Delete flow-typed type definitions

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
Delete .flowconfig

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
eslint: Remove flowtype plugin

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
Start the typescript setup

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
Add @babel/eslint-parser and upgrade to eslint v7

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
eslint: Switch to typescript-eslint parser and such

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
Remove "exact" object type syntax flow typedefs

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
v/transportation/bus: Fix illegal syntax

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

drewvolz commented on .eslintrc.yaml:
```diff
@@ -1,26 +1,24 @@
   - eslint:recommended
-  - plugin:flowtype/recommended
   - plugin:react/recommended
+  - 'plugin:@typescript-eslint/recommended'
```
Chatted with Hawken out of band and he came across [this article](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/TYPED_LINTING.md) on typed linting. 

(It also instructs [how to fix](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/TYPED_LINTING.md#i-get-errors-telling-me-the-file-must-be-included-in-at-least-one-of-the-projects-provided) the two "The file must be included in at least one of the projects provided" errors we currently have for metro.config and babel.config, too.)

Looks like if we enable this plugin inside `.eslintrc.yaml` we gain even more warnings and errors (which is good!)

`1896 problems (1413 errors, 483 warnings)`

```suggestion
  - 'plugin:@typescript-eslint/recommended'
  - 'plugin:@typescript-eslint/recommended-requiring-type-checking'
```

rye replied:
@drewvolz I committed that config change and then uncommitted it; we can always add those lints later, but I think I'd like _valid typescript code_ to be the bar we have to cross in order to merge this PR.

drewvolz replied:
Sounds good to me!

rye committed:
views/views: Fix syntax of ViewType and import of EntypoGlyphs

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
v/student-orgs/list: Split type import to fix syntax

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
eslint: Stick with default --max-warnings

While it is nice sometimes to have all warnings be show-stoppers, in our
case right now it's better to just have the errors be show-stoppers.  As
long as the code can _compile_ (meaning it is at least mostly valid in
the eyes of the TS compiler) we can move forward.  We can look at being
more strict and forcing things to be errors later.  (E.g. via the config
file.)

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
package.json: Don't lint root-level js

The only thing this seems to leave off is dangerfile.js, but otherwise
eslint gets cranky if we give it files that the tsconfig excludes.  We
can add dangerfile.js back into this list if we want to lint it again.

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
m/add-to-device-calendar: Simplify some type syntax to make tsc happy

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
m/filter: Fix typescript-y syntax

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
m/button: Fix type import

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
m/api: Don't use Object explicitly

This type annotation doesn't really do anything, so I replaced it with
unknown.

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
m/ccc-calendar: Fix type import syntax

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
m/datepicker: Replace Object type with any

We can make it more specific later.

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
m/event-list: Fix some type annotations and imports

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
m/fetch: Fix some type annotations and signatures

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
m/searchbar: Fix some type imports

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
m/tableview: Fix type import

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
m/toolbar: Fix a type import

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
m/lists: Fix a type import

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
m/food-menu: Fix a type import

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
tsconfig: Ignore ignored JS files

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
v/settings: Fix type imports

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
s/redux: Fix type imports

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
navigation/track: Fix type import

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
m/food-menu: Fix some more type assertions

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
v/sis: Fix some type imports

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
v/sis: Fix some type assertions

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
v/s/course-search: Fix type imports

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
v/streaming/radio: Fix type imports

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
v/stoprint: Fix some type imports

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
v/building-hours: Fix some type imports

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
Clean up some scattered type imports

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
tsconfig: Fix styling

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
v/s/c-s/lib: Remove // @flow annotations

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
components: Remove // @flow annotations

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
components: Add explicit types on module boundaries

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
components: Replace .js with .ts

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
components: Make non-constants constant

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
i/data: Replace let in for with const

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
init: Remove // @flow annotations

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
init: Rename .js files to .ts

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
components: Add a comment to disambiguate empty arrow functions

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
i/constants: Fix import of version and name from package.json

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
l/course-search: Rename .js files to .ts

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
l/course-search: Remove // @flow annotations

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
l/course-search: Replace let with const where appropriate

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
l/course-search: Add missing module boundary function return type

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
l/financials: Remove // @flow annotations

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
l/financials: Rename .js files to .ts

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
l/financials: Replace let with const where appropriate

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
l/stoprint: Rename .ts files to .js

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
l/stoprint: Remove // @flow annotations

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
l/stoprint/api: Replace ?{} type with ?Record<string, unknown>

This more accurately allows us to express the type we expect; that is,
a type with keys of type string and values of unknown type.

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
l/stoprint: Replace let with const where appropriate

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
l/stoprint/api: Tighten jobId type to string

In other cases, we're using jobId: string, so might as well here, too.

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
l/stoprint: Add return types on boundary functions

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
l/stoprint/errors: Change return type of onDismiss to match Alert param

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
lib: Rename .js files to .ts

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
lib: Remove // @flow annotations

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
lib: Replace let with const where appropriate

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
lib: Add explicit return types to module boundaries

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
navigation: Rename .js files to .ts

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
navigation: Remove // @flow annotations

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
redux: Rename .js files to .ts

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
navigation: Replace let with const where appropriate

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
redux: Remove // @flow annotations

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
navigation: Add an explicit return type to module boundary function

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
redux: Fix parsing errors

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
redux: Replace let with const where appropriate

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
redux: Replace any's and types

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
Remove wonky ?-prefixed types

Either mark as optional, remove, or extend with other possible types.

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
redux: Remove weird type parameters on makeStore

Store doesn't require any type parameters, and this seems to work just
fine after all.

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
Rename app.js to app.tsx

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
components: Reformat

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
m/add-to-device-calendar: Remove // @flow annotations

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
m/add-to-device-calendar: Convert to typescript

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
lib: Reformat

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
l/financials: Reformat

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
l/course-search: Reformat

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
redux: Reformat

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
m/api: Remove // @flow annotation

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
m/add-to-device-calendar: Reformat

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
m/api: Convert to typescript

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
m/app-theme: Convert to typescript

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
m/app-theme: Reformat

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
m/badge: Convert to typescript

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
m/button: Convert to typescript

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
m/ccc-calendar: Convert to typescript

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
m/colors: Convert to typescript

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
m/constants: Convert to typescript

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
m/food-menu: Convert to typescript

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
m/food-menu: Reformat

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
eslint: Mark /.eslintrc.yaml as root

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
eslint: Add a config for scripts/ directory to use old babel parser

Since these scripts need to run on raw Node, we need the syntax to work
on raw Node.  I don't think the ES2015-style imports are available.

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
tsconfig: Enable resolveJsonModule option

This allows us to use import syntax like:

    import {version, name} from '../../package.json'

and have TypeScript resolve '../../package.json' without whining.

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
tsconfig: Ignore scripts/

I don't even want it to try compiling the scripts/ directory at this pt.

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
m/html-content: Convert to typescript

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
m/html-content: Change main entrypoint to index.tsx

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
m/html-lib: Convert to typescript and tighten types

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
Run prettier

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
m/icon: Convert to typescript

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
m/info-header: Convert to typescript

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
package.json: Configure jest to also collect coverage from .ts{,x} files

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
tsconfig: Disallow implicit any type

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
m/layout: Convert to typescript

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
m/lists: Convert to typescript

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
m/markdown: Convert to typescript

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
m/navigation-buttons: Convert to typescript

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
m/navigation-tabs: Convert to typescript

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
m/datepicker: Convert to typescript

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
Merge branch 'master' into nuke-flow

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
Prettify

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
Remove // @flow annotation from everything

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
m/event-list: Rename to .ts{,x} as appropriate

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
eslint: Temporarily suppress prefer-const warnings

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
m/event-list: Fix up formatting of .ts{,x}

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
eslint: Temporarily quiet down two lints

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
m/event-list: Correct the "main" value

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
m/searchbar: Convert to typescript

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
m/storage: Convert to typescript

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
m/tableview: Convert to typescript

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
m/timer: Convert to typescript

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
m/filter: Convert to typescript

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
m/toolbar: Convert to typescript

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
m/notice: Convert to typescript

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
m/open-url: Convert to typescript

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
m/fetch: Convert to typescript

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
m/event-type: Convert to typescript

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
m/separator: Convert to typescript

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
m/silly-card: Convert to typescript

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
m/touchable: Convert to typescript

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
m/use-debounce: Convert to typescript

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
m/viewport: Convert to typescript

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
Fix a gnarly bundling error re: : ? vs ?:

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
v/building-hours/lib: Convert to typescript

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
v/building-hours/lib: Convert tests to typescript as well

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
m/food-menu: Fix an error if mealIndex === -1

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
m/event-list: Fix jest error

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
v/building-hours/lib: Convert get-schedule-status to typescript also

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
v/streaming/radio: Convert to typescript

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
v/building-hours/detail: Convert to typescript

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
root: Convert to typescript

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
v/calendar: Convert to typescript

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
v/faqs: Convert to typescript

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
v/settings/components: Convert to typescript

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
v/settings: Convert to typescript

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
v/sis/components: Convert to typescript

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
v/sis/course-search/detail: Convert to typescript

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
v/{types,views}: Convert to typescript

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
v/menus/lib: Convert to typescript

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
v/building-hours/report: Convert to typescript

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
v/building-hours: Convert to typescript

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
v/dictionary: Convert to typescript

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
v/streaming/streams: Convert to typescript

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
v/streaming/webcams: Convert to typescript

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
v/streaming: Convert to typescript

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
v/transportation/bus: Convert to typescript

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
v/transportation: Convert to typescript

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
v/settings/screens/api-test: Convert to typescript

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
v/settings/screens/debug: Convert to typescript

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
v/settings/screens/overview: Convert to typescript

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
v/settings/screens: Convert to typescript

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
v/settings/screens: Fix some lint errors

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
v/help: Convert to typescript

I also translated the "Object" type into something less opaque!  Yay,
typescript!

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
v/stoprint/components: Convert to typescript

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
v/stoprint: Convert to typescript

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
v/contacts: Convert to typescript

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
v/home: Convert to typescript

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
v/news: Convert to typescript

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
v/menus: Convert to typescript

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
v/student-orgs: Convert to typescript

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
v/sis/course-search/lib: Convert to typescript

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
v/sis/course-search: Convert to typescript

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
v/sis/student-work: Convert to typescript

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
v/sis: Convert to typescript

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
Merge branch 'master' into nuke-flow

There were also some changes that prettier made; I don't think these
should pose a problem.

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
Also cover images/ in the typescript conversion

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
m/fetch: Fix a type error due to failure to return

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
ci: Run yarn tsc to typecheck, goodbye green

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

drewvolz committed:
Swap const for let on a few variables

rye committed:
images: Declare a module for importing jpg files

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
images: Also declare a module for png

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
views: Use a stronger type for gradients

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
m/datepicker: Replace references of ViewStyleProp with ViewStyle

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
m/datepicker: Replace moment type with Moment

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
m/datepicker.android: Fix a couple more tsc issues

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
m/{event-list,layout}: Replace ViewStyleProp references with ViewStyle

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
views: Add optional url string parameter to ViewType

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
views: Downgrade EntypoGlyphs type to just a string

I'm sure there's gotta be a stronger type somewhere, but I'm just not
entirely sure where.

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
modules: Replace a few more ViewStyleProp references

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
Add @types dependencies where appropriate (partial)

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
package: Fix @types/react-navigation-material-bottom-tabs version spec

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
package: Add some typings dependencies

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
tsconfig: Skip checking node_modules/@types

We can't really fix these directly.  It's up to other people to do that,
so avoid checking the node_modules/@types checks entirely.  (Just make a
best effort and extract the types as defined.)

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
yarn: Update the lockfile

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
Prettify

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
Upgrade react-native-popover-view from v1.0.9 to v3.1.1

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
m/add-to-device-calendar: Fix a type mismatch

I believe the goal here was to return a truthy value if promptSettings
returned something truthy, but clearly `Alert | undefined` is not a
boolean.  This commit makes the return value more clear.

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
m/add-to-device-calendar: Correct some return types

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
m/api: Fix API(string, Record<string, any>) signature

While the previous type was technically correct, this commit definitely
makes the types we expect more explicit.  Something Record-y with string
keys _can_ be passed as the second parameter.  If it's not passed, then
it'd be undefined, not null.

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
m/app-theme: Explicitly pass undefined

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
m/button: Replace (.*)BlackObject with \1Object

After closely examining iOSUIKit and material, I figured out that the
"black" objects are actually represented without an additional suffix.

So there are two variants, "calloutWhiteObject" and "calloutObject", of
which the latter is the "black" one.

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
m/button: Remove (incorrect) explicit type declaration

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
m/button: Add undefinedness check

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
Merge branch 'master' into nuke-flow

Conflicts:
	images/webcams/index.js
	- Resolved by taking `oleave` and adding it to the new .ts file.
	package.json
	- Kept the @sentry/cli upgrade.

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
m/ccc-calendar: Fix a number of tsc errors, replace throw with state set

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
Merge branch 'master' into nuke-flow

rye committed:
m/app-theme: Prettify

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
ci: Make `yarn tsc` a soft-fail

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>
